src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -1074,9 +1074,15 @@ namespace ngraph
                 void CPULayout::LAYOUT_DECL(ngraph::op::Result)
                 {
                     auto result = static_cast<const ngraph::op::Result*>(node.get());
-                    if (result->needs_default_layout() ||
-                        mkldnn_utils::get_input_mkldnn_md(node.get(), 0).data.format ==
-                            mkldnn_format_undef)
+                    auto cpu_tvl = dynamic_pointer_cast<runtime::cpu::LayoutDescriptor>(
+                        node->get_inputs()[0]
+                            .get_output()
+                            .get_tensor_view()
+                            ->get_tensor_view_layout());
+
+                    if (result->needs_default_layout() || !cpu_tvl->is_mkldnn_layout() ||
+                        cpu_tvl->get_size() * cpu_tvl->get_element_type().size() !=
```

Comment:
why would sizes be different if it's **not** an mkldnn layout?

Reply:
They wouldn't be.